src/config.ts

```diff
@@ -77,3 +87,89 @@ export const setConfig = (newConfig: ConfigOpts): Config => {
     ...newConfig,
   });
 };
+
+/**
+ * Get the provider and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetProvider = (opts?: ConfigOpts): ethers.providers.Provider => {
+  const c = getConfig(opts);
+  if (!c.provider) throw MissingProvider;
+  return c.provider;
+};
+
+/**
+ * Get the signer and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetSigner = (opts?: ConfigOpts): ethers.Signer => {
+  const c = getConfig(opts);
+  if (!c.signer) throw MissingSigner;
+  return c.signer;
+};
+
+/**
+ * Get the store and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetStore = (opts?: ConfigOpts): StoreInterface => {
+  const c = getConfig(opts);
+  if (!c.store) throw MissingStore;
+  return c.store;
+};
+
+/**
+ * Get the currentFromId and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetCurrentFromId = (opts?: ConfigOpts): string => {
+  const c = getConfig(opts);
+  if (!c.currentFromId) throw MissingUser;
+  return c.currentFromId;
+};
+
+/**
+ * Get the queue.  Since this is a required field, this is a plain getter.
+ * @param opts overrides for the current configuration.
+ */
+export const getQueue = (opts?: ConfigOpts): QueueInterface => {
+  const c = getConfig(opts);
+  return c.queue;
+};
+
+/**
+ * Get the contracts.  Since this is a required field, this is a plain getter.
+ * @param opts overrides for the current configuration.
+ */
+export const getContracts = (opts?: ConfigOpts): Contracts => {
+  const c = getConfig(opts);
+  return c.contracts;
+};
+
+/**
+ * Get the full configuration. Use if you need to guarantee that >1 configs are set
+ * @param requiredConfigs: list of config strings
+ * @param opts
+ */
+export const requireGetConfig = (requiredConfigs?: Array<string>, opts?: ConfigOpts): Config => {
```

Comment:
```
const config = getConfig(opts);

const requiredConfigs = {
  store: MissingStore,
  signer: MissingSigner,
  provider: MissingProvider,
  currentFromId: MissingUser,
}

requiredConfigs
  .filter((configKey) => !config[configKey])
  .forEach( configKey => {
    if (requiredConfigs[configKey]) throw requiredConfigs[configKey];
    
    throw new Error("unknown config: " + configKey);
})

return config;

```

This might be a little easier on the eyeballs if it works. 🤷

Reply:
To me this is more confusing.  Sometimes a long form is better than a short and clever form.